internal/services/compute/image_resource.go

```diff
@@ -513,13 +526,18 @@ func flattenImageDataDisks(input *images.ImageStorageProfile) []interface{} {
 				if disk.ManagedDisk != nil && disk.ManagedDisk.Id != nil {
 					managedDiskId = *disk.ManagedDisk.Id
 				}
+				diskEncryptionSetId := ""
+				if set := disk.DiskEncryptionSet; set != nil && set.Id != nil {
+					diskEncryptionSetId = *set.Id
```

Comment:
We should parse the disk encryption set ID returned by the API here before setting it into state (this is so we can potentially catch any casing inconsistencies in the ID)

Reply:
Done